ground/src/main/java/com/google/android/ground/system/GoogleApiManager.kt

```diff
@@ -37,29 +38,43 @@ constructor(
    * Displays a dialog to install Google Play Services, if missing. Throws an error if install not
    * possible or cancelled.
    */
-  suspend fun installGooglePlayServices() {
+  suspend fun installGooglePlayServices(): Boolean {
     val status = googleApiAvailability.isGooglePlayServicesAvailable(context)
-    if (status == ConnectionResult.SUCCESS) return
+    if (status == ConnectionResult.SUCCESS) return true
 
     val requestCode = INSTALL_API_REQUEST_CODE
-    startResolution(status, requestCode, GooglePlayServicesMissingException())
-    getNextResult(requestCode)
+    return startResolution(status, requestCode)
   }
 
-  private fun startResolution(status: Int, requestCode: Int, throwable: Throwable) {
-    if (!googleApiAvailability.isUserResolvableError(status)) throw throwable
+  private suspend fun startResolution(status: Int, requestCode: Int): Boolean {
+    if (!googleApiAvailability.isUserResolvableError(status)) return false
 
-    activityStreams.withActivity {
-      googleApiAvailability.showErrorDialogFragment(it, status, requestCode) { throw throwable }
+    return try {
+      activityStreams.withActivity { activity ->
+        googleApiAvailability.showErrorDialogFragment(activity, status, requestCode, null)
+      }
+      if (getNextResult(requestCode)) {
+        true
+      } else {
+        Timber.e("Activity result was not successful for requestCode: $requestCode")
+        false
+      }
+    } catch (e: Exception) {
+      Timber.e(
+        e,
+        "Failed to launch app resolution for status: $status and requestCode: $requestCode",
+      )
+      false
     }
   }
 
-  private suspend fun getNextResult(requestCode: Int) {
+  private suspend fun getNextResult(requestCode: Int): Boolean {
     val result = activityStreams.getNextActivityResult(requestCode)
-    if (!result.isOk()) {
-      error("Activity result failed: requestCode = $requestCode, result = $result")
+    return if (result.isOk()) {
+      true
+    } else {
+      Timber.e("Activity result failed: requestCode = $requestCode, result = $result")
```

Comment:
Does this indicate a bug, or an issue with the user's device? If it's the latter, we probably don't want it in error logs and Crashlytics - reducing to Timber.d should be sufficient.


Same comment for other new Timber.e calls.

Reply:
done